### ISSUE TITLE
TypeError on Chrome 51 - observer.thresholds is undefined

### ISSUE THREAD

**Issue by md384** [thebuilder/react-intersection-observer#192]
I am seeing this error `TypeError: Cannot read property 'some' of undefined` for users on Chrome 51
caused by https://github.com/thebuilder/react-intersection-observer/blob/295ae5337411caecfce537c9e5b6cd68027e3ae9/src/intersection.ts#L153-L160

In Chrome 51, the `thresholds` attribute of `IntersectionObserver` is `undefined`, so `thresholds.some` throws the error.
```
// Chrome 51
observer = new IntersectionObserver(() => null, {})
console.log(observer.thresholds) // undefined

// Chrome 52+
observer = new IntersectionObserver(() => null, {})
console.log(observer.thresholds) // [0]
```

Since we only polyfill if `IntersectionObserver` is `undefined`, the polyfill doesn't help in this case. I'm not sure what the best course of action here but I think there should be a check that `thresholds` is defined to avoid this error. Perhaps, `thresholds` could be added to the `ObserverInstance` in `INSTANCE_MAP` and used if `instance.observer.thresholds` is `undefined`.

**Comment by thebuilder** [comment]
Chrome 51 should return the `thresholds` value (https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/thresholds), so not sure what is going on here.

**Comment by thebuilder** [comment]
But I'm also seeing the issue. I'll add the `threshold` to the instance, so it can fall back to that value. if `thresholds` is undefined.

**Comment by thebuilder** [comment]
Solved in #194
